tensorflow/core/kernels/batch_kernels.cc

```diff
@@ -520,10 +520,8 @@ class BatchFunctionKernel : public AsyncOpKernel {
 
 REGISTER_KERNEL_BUILDER(Name("BatchFunction").Device(DEVICE_CPU),
                         BatchFunctionKernel);
-// Currently all inputs and outputs are on the host.
-// TODO(b/173748277): Accept inputs/outputs on the device.
 REGISTER_KERNEL_BUILDER(Name("BatchFunction")
-                            .Device(DEVICE_GPU)
+                            .Device(DEVICE_DEFAULT)
                             .HostMemory("in_tensors")
                             .HostMemory("captured_tensors")
                             .HostMemory("out_tensors"),
```

Comment:
Since there is a plan to accept input/outputs on the device for `DEVICE_GPU`, let's keep this registration (and the comments). Please make a separate one for `DEVICE_DEFAULT`. Thank you!